src/tests/thunder/thunder_loadtest.test.js

```diff
@@ -0,0 +1,74 @@
+import {
+  getControllerPluginData,
+  setWebKitUrl,
+  webKitBrowserStartAndResume,
+} from '../../commonMethods/commonFunctions'
+
+let listener
+
+export default {
+  title: 'Thunder Load test',
+  description: 'Stress tests the Thunder',
+  setup() {
+    return this.$sequence([
+      () => webKitBrowserStartAndResume.call(this),
+      () =>
+        (listener = this.$thunder.api.WebKitBrowser.on('urlchange', data => {
+          this.$data.write('currentUrl', data.url)
+        })),
+    ])
+  },
+  teardown() {
+    listener.dispose()
+  },
+  steps: [
+    {
+      description: 'Navigating to Thunder UI',
+      test() {
+        let host = this.$thunder.api.options.host
+        let url = `http://${host}/Service/Controller/UI?ip=&port=65535`
+        this.$data.write('url', url)
+        return setWebKitUrl.call(this, url)
+      },
+      validate(url) {
+        return url === this.$data.read('url')
+      },
+    },
+    {
+      description: 'Sleep until URL is loaded',
+      sleep() {
+        // Purpose of this sleep is to wait until current step gets 'url change' response from the listener
+        return new Promise((resolve, reject) => {
+          let attempts = 0
+          const interval = setInterval(() => {
+            attempts++
+            if (this.$data.read('currentUrl') === this.$data.read('url')) {
+              clearInterval(interval)
+              resolve()
+            } else if (attempts > 10) {
+              clearInterval(interval)
+              reject('URL not loaded within time limit')
+            }
+          }, 1000)
+        })
+      },
+    },
+    {
+      title: 'Repeat the steps for 1 hour',
+      description: 'Nested test to repeat the test for 1 hour',
+      repeat: {
+        seconds: 1 * 60 * 60, //One hour
+      },
```

Comment:
You need to keep reloading thunder UI during that one hour repeat. So if you move the first 2 steps within the repeat the test is great!

Reply:
@wouterlucas , 
Updated the code as per the above comment. Please review and let me know your feedback if any

Reply:
Great! Thanks, just one remark when we set the url again does the browser actually reload the page? or do we navigate away and back to force a full refresh of the UI?

Reply:
@wouterlucas , Browser loads full fresh ThunderUI(the URl we set).

Reply:
Thanks! Perfect.